bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts

```diff
@@ -207,26 +212,47 @@ export class OverviewComponent implements OnInit, OnDestroy {
         next: async (view) => {
           this.showOnboarding = Object.values(view.tasks).includes(false);
           if (this.organizationId) {
-            await this.identifyOrganizationsWithUpcomingPaymentIssues();
+            await this.locateOrganizationsWithIncomingPaymentIssues();
           }
           this.loading = false;
         },
       });
   }
 
-  private async identifyOrganizationsWithUpcomingPaymentIssues() {
-    const sub = await this.organizationApiService.getSubscription(this.organizationId);
-    const billing = await this.organizationApiService.getBilling(this.organizationId);
-    const trialEndDate = sub?.subscription?.trialEndDate;
-    this.isOwner = this.organization.isOwner;
-    this.isTrailing = sub.subscription.status == "trialing";
-    this.defaultPaymentSource = billing;
-    if (trialEndDate) {
-      const today = new Date();
-      const trialEnd = new Date(trialEndDate);
-      const timeDifference = trialEnd.getTime() - today.getTime();
-      this.trialRemainingDays = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
-    }
+  private async locateOrganizationsWithIncomingPaymentIssues() {
+    combineLatest([
+      from(this.organizationApiService.getSubscription(this.organizationId)),
+      from(this.organizationApiService.getBilling(this.organizationId)),
+    ])
+      .pipe(
+        tap(([sub, billing]) => {
+          const trialEndDate = sub?.subscription?.trialEndDate;
+          this.isOwner = this.organization?.isOwner;
+          this.isTrialing = sub?.subscription?.status == "trialing";
+          this.defaultPaymentSource = billing;
+          if (trialEndDate) {
+            const today = new Date();
+            const trialEnd = new Date(trialEndDate);
+            const timeDifference = trialEnd.getTime() - today.getTime();
+            this.trialRemainingDays = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
+            this.updateFreeTrialMsg();
+          }
+        }),
+        takeUntil(this.destroy$),
+      )
+      .subscribe();
+  }
+
+  updateFreeTrialMsg() {
+    this.freeTrialMessage = computed(() => {
+      if (this.trialRemainingDays >= 2) {
+        return this.i18nService.t("freeTrialEndPrompt", this.trialRemainingDays);
+      } else if (this.trialRemainingDays == 1) {
+        return this.i18nService.t("freeTrialEndPromptForOneDayNoOrgName");
+      } else {
+        return this.i18nService.t("freeTrialEndingSoonWithoutOrgName");
+      }
+    });
   }
```

Comment:
The [computed](https://v16.angular.io/api/core/computed) is in `Developer preview`. It was released in v17+, but why not just use angular natively supported computed getter ?
```suggestion
  get freeTrialMessage() {
      if (this.trialRemainingDays >= 2) {
        return this.i18nService.t("freeTrialEndPrompt", this.trialRemainingDays);
      } else if (this.trialRemainingDays == 1) {
        return this.i18nService.t("freeTrialEndPromptForOneDayNoOrgName");
      } else {
        return this.i18nService.t("freeTrialEndingSoonWithoutOrgName");
      }
  }
```